### PR TITLE
Update/redirect js lib params

### PR DESCRIPTION
**Pull request by leogermani**
This PR updates the Redirect JS lib to make it simpler and accept and pass through any parameter instead of having a controlled list.

This allows us to cover all situations where redirect links are created. It also makes it easier to add new params to the URL.

Also, this PR correctly passes the `unliked` param inside the `query` as this is not used by the redirect server and we only need to send it to the final destination URL

#### Changes proposed in this Pull Request:

* Improves Jetpack Redirect package by accepting any parameter and making the docs better descriptive. 

#### Jetpack product discussion

p9dueE-36H-p2

#### Testing instructions:

TBD

### CODE REVIEW

File: projects/plugins/jetpack/_inc/client/state/initial-state/reducer.js [hunk 1]
```diff
@@ -395,16 +396,30 @@ export const getUpgradeUrl = ( state, source, userId = '', planDuration = false
...
+		redirectArgs.u = uid;
+	}
+	if ( subsidiaryId ) {
+		redirectArgs.subsidiaryId = subsidiaryId;
+	}
+	if ( ! isCurrentUserLinked( state ) ) {
+		redirectArgs.query = 'unlinked=1';
+	}
+	if ( purchaseToken ) {
+		redirectArgs.purchasetoken = purchaseToken;
```
Comment (sergeymitr) [inline projects/plugins/jetpack/_inc/client/state/initial-state/reducer.js:416]: Is `purchasetoken` a separate argument, or we should put that into `query` as well?
On the "Plans" link it does go into the `query`: https://github.com/Automattic/jetpack/blob/master/projects/plugins/jetpack/_inc/client/components/navigation/index.jsx#L60

Comment (leogermani) [reply projects/plugins/jetpack/_inc/client/state/initial-state/reducer.js:416]: Yes, I highlighted it in the p2 and was waiting for the discussion [here](p9dueE-36H-p2#comment-5499) to decide. I'll update the PR with the decision.

Comment (leogermani) [reply projects/plugins/jetpack/_inc/client/state/initial-state/reducer.js]: done

**Comment by leogermani** [comment]
I don't think there's any testable scenario here because we are not displaying any checkout link to a not connected user.

**Review by jeherve — APPROVED** [review]

**Pull request merged**